content/api/loaders.md

```diff
@@ -93,33 +104,60 @@ In `/abc/file.js`:
 require("./loader1?xyz!loader2!./resource?rrr");
 ```
 
-### `version`
+### `this.version`
 
 **Loader API version.** Currently `2`. This is useful for providing backwards compatibility. Using the version you can specify custom logic or fallbacks for breaking changes.  
 
-### `context`
+### `this.context`
 
 **The directory of the module.** Can be used as context for resolving other stuff.
 
 In the example: `/abc` because `resource.js` is in this directory
 
-### `request`
+### `this.request`
 
 The resolved request string.
 
 In the example: `"/abc/loader1.js?xyz!/abc/node_modules/loader2/index.js!/abc/resource.js?rrr"`
 
-### `query`
+### `this.query`
+
+1. In case the loader was configured with an [`options`](/configuration/module/#useentry) object, this will be a reference to the object.
+2. If the loader has no `options`, but was invoked with a query string, this will be a string starting with `?`.
+
+Use the [`getOptions` method from the `loader-utils`](https://github.com/webpack/loader-utils#getoptions) to extract the given loader options.
+
+### `this.callback`
 
-A string. The query of the request for the current loader.
+A function that can be called synchronously or asynchronously in order to return multiple results. The expected arguments are:
 
-In the example: in loader1: `"?xyz"`, in loader2: `""`
+```typescript
+this.callback(
+    err: Error | null,
+    content: string | Buffer,
+    sourceMap?: SourceMap,
+    abstractSyntaxTree?: AST
+);
+```
+
+1. The first argument must be an `Error` or `null`
+2. The second argument a `string` or a [`Buffer`](https://nodejs.org/api/buffer.html).
+3. Optional: The third argument must be a source map that is parsable by [this module](https://github.com/mozilla/source-map).
+4. Optional: `AST` can be an Abstract Syntax Tree of the given language, like [`ESTree`](https://github.com/estree/estree). This value is ignored by webpack itself, but useful to speed up the build time if you want to share common ASTs between loaders.
+
+In case this function is called, you should return undefined to avoid ambigious loader results.
+
+### `this.async`
 
-### `data`
+A function than returns either `callback` or a falsy value depending whether asynchronous loader results are supported by the loader pipeline or not.
```

Comment:
There's actually a typo at `than`.

I would invert the problem and go with something like this:

The results of `this.async` depend on the loader pipeline. If the pipeline supports asynchronous behavior, it returns a callback. Otherwise it returns a falsy value.

---

Then refer to that loader pipeline idea so people can check out what that means. Now there isn't enough context. Ideally you would have two different examples too to cover all the bases.